File: functions/services/openai-service.js
const functions = require('firebase-functions');
const OpenAI = require('openai');

/**
 * OpenAI Service
 */
class OpenAIService {
  constructor() {
    this.apiKey = functions.config().openai?.api_key;
    this.modelName = 'gpt-4';
    
    if (!this.apiKey) {
      console.warn('OPENAI_API_KEY not configured');
    }
    
    if (this.apiKey) {
      this.client = new OpenAI({
        apiKey: this.apiKey
      });
    }
  }

  async generate(prompt) {
    if (!this.client) {
      throw new Error('OpenAI API Key未配置');
    }

    try {
      const completion = await this.client.chat.completions.create({
        model: this.modelName,
        messages: [
          { role: 'user', content: prompt }
        ],
        temperature: 0.7,
        max_tokens: 800
      });

      return completion.choices[0].message.content;

    } catch (error) {
      console.error('OpenAI generation error:', error);
      throw new Error(`OpenAI生成失败: ${error.message}`);
    }
  }

  getModelName() {
    return this.modelName;
  }
}

module.exports = OpenAIService;

File: backend/services/openai_service.py
"""
OpenAI服务实现
"""

import os
import logging
from openai import OpenAI

from services import AIService

logger = logging.getLogger(__name__)


class OpenAIService(AIService):
    """OpenAI服务实现"""
    
    def __init__(self):
        self.api_key = os.getenv('OPENAI_API_KEY')
        self.model_name = 'gpt-4'
        self.client = None
        
        if self.api_key:
            try:
                self.client = OpenAI(api_key=self.api_key)
                logger.info("OpenAI service initialized successfully")
            except Exception as e:
                logger.error(f"Failed to initialize OpenAI: {e}")
        else:
            logger.warning("OPENAI_API_KEY not found in environment")
    
    def generate(self, prompt: str) -> str:
        """使用OpenAI生成文本"""
        if not self.client:
            raise ValueError("OpenAI service is not properly initialized")
        
        try:
            response = self.client.chat.completions.create(
                model=self.model_name,
                messages=[
                    {"role": "user", "content": prompt}
                ],
                temperature=0.7,
                max_tokens=800
            )
            
            return response.choices[0].message.content
            
        except Exception as e:
            logger.error(f"OpenAI generation error: {e}")
            raise Exception(f"OpenAI生成失败: {str(e)}")
    
    def is_available(self) -> bool:
        """检查OpenAI服务是否可用"""
        return self.client is not None and self.api_key is not None
    
    def get_model_name(self) -> str:
        """获取模型名称"""
        return self.model_name
